File: Data Structures & Algorithms/is-anagram/submission-2.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s)!=len(t):
            return False;
        count=[0]*26;
        for i,j in zip(s,t):
            count[ord(i)-97]+=1
            count[ord(j)-97]-=1
        return count==[0]*26
            

File: Data Structures & Algorithms/anagram-groups/submission-2.rs
use std::collections::HashMap;

impl Solution {
    pub fn group_anagrams(strs: Vec<String>) -> Vec<Vec<String>> {
        let mut groups:HashMap<[u8;26],Vec<String>>=HashMap::new();

        for word in strs{
            let mut count=[0;26];
            for ch in word.as_bytes(){
                count[(ch-b'a') as usize]+=1;
            }
            groups.entry(count).or_insert(Vec::new()).push(word);
        }
        groups.into_values().collect()
    }
}